tests/test_gui.py

```diff
@@ -0,0 +1,239 @@
+        "Show/hide line numbers."
+    ].exists(), "Show/hide line numbers checkbox is missing"
+    assert app_dialog.root.buttons[
+        "Add a new tab"
+    ].exists(), "Add a new tab button is missing"
+    assert app_dialog.root.tabs[".py"].exists(), ".py tab is missing"
+
+
+def test_load_script(app_dialog):
+    """
+    Make sure Python Console can load and run a python script
+    """
+    # Load a script
+    app_dialog.root["Load python script from a file."].mouseClick()
+    app_dialog.root.dialogs["Open Python Script"].waitExist(timeout=30)
+    open_script_path = os.path.normpath(
+        os.path.expandvars("${TK_TEST_FIXTURES}/files/script/UiAutomationScript.py")
+    )
+    app_dialog.root.dialogs["Open Python Script"].textfields["File name:"].pasteIn(
+        open_script_path, enter=True
+    )
+    app_dialog.root.tabs["UiAutomationScript.py"].waitExist()
+    # Clear output dialog and disable echo python commands
+    app_dialog.root.buttons["Clear all output."].mouseClick()
+    assert app_dialog.root.captions[""].exists()
+    app_dialog.root.checkboxes["Echo python commands in output."].mouseClick()
+    assert (
+        app_dialog.root.checkboxes["Echo python commands in output."].selected is False
+    ), "Echo python commands in output checkbox should be disabled"
+    # Run the loaded script
+    app_dialog.root.buttons[
+        "Execute the current python script. Shortcut: Ctrl+Enter"
+    ].mouseClick()
+    assert app_dialog.root.captions[
+        "0\n1\n2"
+    ].exists(), "Print Ci Automation is missing"
+
+
+def test_save_script(app_dialog):
+    """
+    Make sure Python Console can save a python script
+    """
+    # Crate a new tab, run and save a Python script
+    app_dialog.root.buttons["Clear all input."].mouseClick()
+    app_dialog.root.buttons["Add a new tab"].mouseClick()
+    app_dialog.root.textfields.mouseClick()
+    app_dialog.root.textfields.pasteIn('print("Hello World!")')
+    app_dialog.root.checkboxes["Show/hide line numbers."].mouseClick()
+    app_dialog.root.buttons[
+        "Execute the current python script. Shortcut: Ctrl+Enter"
+    ].mouseClick()
+    assert app_dialog.root.captions[
+        '*print("Hello World!")\nHello World!'
+    ].exists(), "Print Hello World! is missing"
+    # Save the updated script
+    app_dialog.root.buttons["Save current python script to a file."].mouseClick()
+    app_dialog.root.dialogs["Save Python Script"].waitExist(timeout=30)
+    save_script_path = os.path.normpath(
+        os.path.expandvars(
+            "${TK_TEST_FIXTURES}/files/script/UiAutomationScriptUpdated.py"
+        )
+    )
+    app_dialog.root.dialogs["Save Python Script"].textfields["File name:"].pasteIn(
+        save_script_path, enter=True
+    )
+    # Validate the file exist locally
+    assert os.path.isfile(
+        os.path.expandvars(
+            "${TK_TEST_FIXTURES}/files/script/UiAutomationScriptUpdated.py"
+        )
+    )
```

Comment:
small thing but I wonder if we could read back in the contents of the file to make sure it's writing correctly?

Reply:
Good point! Fixed